test/e2e/network/service.go

```diff
@@ -2741,6 +2752,13 @@ var _ = common.SIGDescribe("Services", func() {
 		nodePortAddress0 := net.JoinHostPort(nodeIPs0[0], strconv.Itoa(int(svc.Spec.Ports[0].NodePort)))
 		nodePortAddress1 := net.JoinHostPort(nodeIPs1[0], strconv.Itoa(int(svc.Spec.Ports[0].NodePort)))
 		for i := 0; i < 5; i++ {
+			// There's a terminating pod called webserver0 running on node0.
+			// There's a Service Type=NodePort with externalTrafficPolicy=Local
+			// pausePod0 and pausePod1 are on node0 and node1 respectively.
+			// pausePod0 -> node1 nort port fails because it's "external" and there are no local endpoints
+			// pausePod0 -> node0 node port succeeds because webserver0 is running on node0
+			// pausePod1 -> node0 node port succeeds because webserver0 is running on node0
+			// pausePod1 -> node1 node port succeeds because this is considered "internal"
```

Comment:
`pausePod0 -> node0` and `pausePod1 -> node1` both succeed because pod-to-same-node-NodePort connections are neither internal nor external and always get Cluster traffic policy.